src/scene/scene.c

```diff
@@ -623,6 +623,31 @@ static int sceneUpdateTurret(struct Scene* scene, void* object) {
     return 1;
 }
 
+void sceneUpdateCameraZoom(struct Scene* scene) {
+    if (playerIsDead(&scene->player)) {
+        scene->camera.fov = DEFAULT_CAMERA_FOV;
+        scene->portalGun.fov = DEFAULT_PORTALGUN_FOV;
+        return;
+    }
+
+    if (scene->player.zoomTimer >= ZOOM_CAMERA_TIME) {
+        return;
+    }
+
+    scene->player.zoomTimer += FIXED_DELTA_TIME;
```

Comment:
This can exceed `ZOOM_CAMERA_TIME` by a bit and so the FOVs will be slightly off.

Please use `clampf()` to clamp the lerp amount to [0, 1] before passing to `mathfLerp()`.